torch/csrc/jit/ir.h

```diff
@@ -972,6 +973,11 @@ struct Graph {
   std::unordered_set<const Block*> all_blocks;
   size_t next_unique_;
 
+  // This is used to identify instance of a graph uniquely within a process.
+  // This is used for source range augmentation in profiling that generates
+  // a trace.
+  int64_t graph_id_{-1};
```

Comment:
it feels like the wrong place to tie things together. As commented above, I'd rather keep some pointer to CodeImpl or Graph itself inside profiler (we can downcast it to shared_ptr<void> if needed) instead of polluting JIT code

Reply:
@dzhulgakov, so every event needs two pieces of information. 1) which code and 2) what pc. Earlier I did 1) by storing graph_id_, but this is wrong, so instead I need something equivalent of code_id. Trouble with storing shared_ptr<CodeImpl> for every event is that you will need to store shared_ptr which will involve atomic counters and this is what I wanted to avoid for per reason. I think it is possible to keep all the code inside interpreter.cpp.

1. Each CodeImpl instance has a unique id and source_debug_info maintains a map that maps from this unique id to shared_ptr<CodeImpl>. This is done each time Code is generated, likely at the creation of ExecutionPlan.
2. During the execution then each even is annotated by CodeImpl instance's unique id and the pc.

Note I still feel stringifying debug info at the time of instruction insertion is better than doing at the time of trace creation since we will be creating and destroying lot of string otherwise.

Reply:
Also I realize CodeImpl or Code is part of interpreter, so storing a shared pointer to it requires including `interpreter.h`. This I feel is not appropriate as profiling does not need access to interpreter.h.